### PR TITLE
improve error management issue#30

### PR DESCRIPTION
**Pull request by alimand**
Add pop up error message box if GDC is not working as expectation and avoid stuck of the application

### CODE REVIEW

File: src/frontend/components/CatalogueView.vue [hunk 1]
```diff
@@ -247,95 +247,104 @@ export default defineComponent({
...
-            if (searchedTitle.value) {
-                params.append('title', searchedTitle.value);
-            }
-            // Add query if it exists
-            if (query.value) {
-                params.append('q', query.value);
-            }
+// Inside the searchCatalogue method
+// Search the catalogue
+const searchCatalogue = async () => {
```
Comment (RoryPTB) [inline src/frontend/components/CatalogueView.vue]: This indention is incorrect

**Review by RoryPTB — CHANGES_REQUESTED** [review]
Since you've move the entire `searchCatalogue` method into a try block, most of the changes here are just indentation changes. 

You don't need to wrap the entire method in a try block - instead, take the code from `const response ...` onwards and move it to a separate helper function, which we could call `fetchAPI` for example. Then, in the `searchCatalogue` method you can call this function and wrap it in a try/catch/finally block. This will make the code easier to understand.

File: src/frontend/components/CatalogueView.vue (hunk 1 above)
Comment (alimand) [reply src/frontend/components/CatalogueView.vue]: Thanks for figure out that problem

**Comment by alimand** [comment]
> Since you've move the entire `searchCatalogue` method into a try block, most of the changes here are just indentation changes.
> 
> You don't need to wrap the entire method in a try block - instead, take the code from `const response ...` onwards and move it to a separate helper function, which we could call `fetchAPI` for example. Then, in the `searchCatalogue` method you can call this function and wrap it in a try/catch/finally block. This will make the code easier to understand.

@RoryPTB I tried to make a fetchAPI to handle the searchCatalogue method and make this update

**Pull request closed without merging**